port/common/omrstr.c

```diff
@@ -2195,11 +2199,24 @@ omrstr_current_time_zone(struct OMRPortLibrary *portLibrary, int32_t *secondsEas
 		if (NULL != secondsEast) {
 			*secondsEast = zoneSecondsEast;
 		}
-		if (NULL != zoneNameBuffer) {
-			strncpy(zoneNameBuffer, zoneName, zoneNameBufferLen);
-			if (0 != zoneNameBufferLen) {
-				zoneNameBuffer[zoneNameBufferLen - 1] = '\0';
+		if ((NULL != zoneNameBuffer) && (NULL != zoneName)) {
+			size_t fullLength = strlen(zoneName);
+			size_t trimLength = (fullLength < zoneNameBufferLen) ? fullLength : (zoneNameBufferLen - 1);
```

Comment:
In case of a non-null `zoneNameBuffer` with `zoneNameBufferLen = 0`, will the new code function correctly?

Reply:
Yes, it writes an empty string to `zoneNameBuffer`.

Reply:
When `zoneNameBufferLen = 0`, a negative value will be stored in `trimLength` which is unsigned: `trimLength = 0 - 1 = -1`. There needs to be a better guard:
```
if ((NULL != zoneNameBuffer) && (NULL != zoneName) && (0 != zoneNameBufferLen)) {
```

Reply:
Fixed.